### ISSUE TITLE
Double anchor problem with 'singlehtml' Builder

### ISSUE THREAD

**Issue by bigbes** [sphinx-doc/sphinx#2363]
Firstly appeared in https://github.com/sphinx-doc/sphinx/issues/151#issuecomment-68528760 this comment.

There's a function [`SingleFileHTMLBuilder.fix_refuris`](https://github.com/sphinx-doc/sphinx/blob/master/sphinx/builders/html.py#L936) that must fix this issue, but it doesn't work on TOC directive:

```
.. toctree::
```

We've reproduced it in Tarantool project [documentation](http://tarantool.org/doc/singlehtml.html) [singlehtml](https://raw.githubusercontent.com/tarantool/tarantool/1.6/doc/sphinx/singlehtml.rst) file.

```
<ul>
<li class="toctree-l1"><a class="reference internal" href="singlehtml.html#document-intro">Overview</a><ul>
<li class="toctree-l2"><a class="reference internal" href="singlehtml.html#document-intro#key-features">Key features</a></li>
<li class="toctree-l2"><a class="reference internal" href="singlehtml.html#document-intro#how-stable-is-the-software">How stable is the software?</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="singlehtml.html#document-faq">FAQ</a></li>
<li class="toctree-l1"><a class="reference internal" href="singlehtml.html#document-getting_started">Getting started</a><ul>
<li class="toctree-l2"><a class="reference internal" href="singlehtml.html#document-getting_started#downloading-and-installing-a-binary-package">Downloading and installing a binary package</a></li>
<li class="toctree-l2"><a class="reference internal" href="singlehtml.html#document-getting_started#starting-tarantool-and-making-your-first-database">Starting Tarantool and making your first database</a></li>
<li class="toctree-l2"><a class="reference internal" href="singlehtml.html#document-getting_started#starting-another-tarantool-instance-and-connecting-remotely">Starting another Tarantool instance and connecting remotely</a></li>
</ul>
```

I'll try to reproduce this problem in a small test-case.

**Comment by bigbes** [comment]
Problem was in function [`toctree()`](http://www.sphinx-doc.org/en/stable/templating.html#toctree) in my templates. I've replaced it with `{{ toc }}` and double anchors have disappeared.

**Comment by bigbes** [comment]
Test case is here: https://github.com/bigbes/sphinx-testcase

To repeat: 

```
sphinx-build -b singlehtml . _output
open _output/index.html
```

and try to click on any link on level 2

**Comment by tk0miya** [comment]
Thank you for reporting. Fixed at f8c08c6.

**Comment by bigbes** [comment]
Thank you!
